### PR TITLE
stlink: set darwin badPlatforms 

### PR DESCRIPTION
**Pull request by ocfox**
## Description of changes
see https://hydra.nixos.org/build/249202776

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by wegank — DISMISSED** [review]

**Comment by bjornfor** [comment]
Can you please add a comment about the removed patch (in the commit message)?

**Comment by ocfox** [comment]
Added it.

**Comment by bjornfor** [comment]
> Added it.

Thanks, but that comment didn't add the kind of context I was looking for. After figuring out the situation myself, I'd say something like `Remove patch that is included in version 1.8.0 and now fails to apply.`.

(Alternative, the build fix could be done by reverting 49dce7182fe550d190b73ede64295c486c1b6763.)

**Comment by bjornfor** [comment]
I first read the "fix build" topic as referring to  opening up meta.platforms. Hence my confusion about the meaning of the removed patch (since it wasn't mentioned in the commit message, and I didn't expect last stlink PR to have broken to build).

**Comment by ocfox** [comment]
So how do you think the change will be solved? Keep a 1.7.0 with patch or completely cancel the support for darwin.

**Comment by bjornfor** [comment]
> So how do you think the change will be solved? Keep a 1.7.0 with patch or completely cancel the support for darwin.

I don't understand. All I'm saying is that I got confused about the diff vs the commit message. (And it's logically doing two different things: fixing the build by removing outdated patch (for all platforms) and updating meta.platforms to re-allow unix - darwin.)

File: pkgs/development/tools/misc/stlink/default.nix [hunk 1]
```diff
@@ -55,8 +48,8 @@ in stdenv.mkDerivation rec {
   meta = with lib; {
     description = "In-circuit debug and programming for ST-Link devices";
     license = licenses.bsd3;
-    # upstream says windows, linux/unix but dropped support for macOS in 1.8.0
-    platforms = platforms.linux; # not sure how to express unix without darwin
+    platforms = platforms.unix;
+    badPlatforms = platforms.darwin;
```
Comment (ocfox) [inline pkgs/development/tools/misc/stlink/default.nix:52]: The change is "express unix without darwin" and does not re-allow darwin.

Comment (evils) [reply pkgs/development/tools/misc/stlink/default.nix:52]: `platform.unix` includes `darwin` which upstream explicitly states is no longer supported...

Comment (ocfox) [reply pkgs/development/tools/misc/stlink/default.nix:52]: See https://nixos.org/manual/nixpkgs/unstable/#var-meta-badPlatforms.

Comment (evils) [reply pkgs/development/tools/misc/stlink/default.nix:52]: nice, thanks!

**Comment by bjornfor** [comment]
@ocfox: https://github.com/NixOS/nixpkgs/pull/287987 was merged, so now you can focus on the "meta" part of the commit.

**Review by bjornfor — APPROVED** [review]

**Review by evils — APPROVED** [review]

**Comment by bjornfor** [comment]
@ocfox: thanks!

**Pull request merged**